### ISSUE TITLE
Manifest V3 pack dependencies with alpha/beta suffixes throw false positive "Incorrect type. Expected array"

### ISSUE THREAD

**Issue by capuchin8** [Blockception/Minecraft-bedrock-json-schemas#626]
**Description of the bug**
When using `format_version: 3` in the `manifest.json`, the extension throws a false positive validation warning in the `dependencies` block: `Incorrect type. Expected "array"`. 

Crucially, this warning **only** occurs when a pack dependency uses a SemVer string containing pre-release identifiers, such as alpha, beta or RC tags (e.g., `"1.21.0-beta"`, `"1.21.0-alpha"` or `"1.21.0-rc"`). Standard module versions and purely numeric version strings (e.g., `"1.10.0"`) pass validation without issue. 

If a developer changes the version to an integer array to satisfy the validator, Minecraft completely fails to load the pack, as Manifest V3 strictly requires these to be strings.

**To Reproduce**

1. Create a `manifest.json` file with `"format_version": 3`.
2. Add a pack dependency using a pre-release/beta string identifier:
```json
"dependencies": [
  {
    "uuid": "example-pack-uuid-here",
    "version": "1.21.0-beta"
  }
]
```
3. Observe the validation linting error: `Incorrect type. Expected "array"`.
4. Change the version to a standard numeric string (e.g., `"1.21.0"`) or reference a module, and observe that the error disappears.

**Expected behavior**
The JSON schema parser should recognize that Manifest V3 string format targets for `version` fields can legally include SemVer-compliant pre-release labels (like `-beta`, `-alpha` or `-rc`) without failing back to the V2 array assumption.

**Desktop (please complete the following information):**

- OS: Windows 11 22H2
- Version 9.1.17
